snapshots/testsuite/testsuite.go

```diff
@@ -149,12 +149,12 @@ func checkSnapshotterBasic(ctx context.Context, t *testing.T, snapshotter snapsh
 	if err := mount.All(mounts, preparing); err != nil {
 		t.Fatalf("failure reason: %+v", err)
 	}
-	defer testutil.Unmount(t, preparing)
 
 	if err := initialApplier.Apply(preparing); err != nil {
 		t.Fatalf("failure reason: %+v", err)
 	}
 
+	testutil.Unmount(t, preparing)
```

Comment:
Can you put this before the error `Apply` error check to ensure it gets called on failure too

Reply:
Good point! Turns out there are a few failure cases that need cleanup. I'll fix it. Thanks!